puppet/zulip/manifests/app_frontend_base.pp

```diff
@@ -131,7 +131,11 @@
   # This determines whether we run queue processors multithreaded or
   # multiprocess.  Multiprocess scales much better, but requires more
   # RAM; we just auto-detect based on available system RAM.
-  $queues_multiprocess_default = $zulip::common::total_memory_mb > 3500
+  #
+  # We use a cutoff less than 4000 here to detect "4GB" systems,
+  # because some may have as little as 4 x 1000^3 / 1024^3 = 3714 MiB
+  # of memory.
+  $queues_multiprocess_default = $zulip::common::total_memory_mb > 3800
```

Comment:
This is the first reference for "4GB" in the file, so some preamble referencing that we consider < 4GB to be a "low memory" environment would be useful context.

4*(1000^3)/(1024^3) = 3.7252902985, not 3.714?

I'm a bit confused why you show that the lower limit for "4GB" systems is 3714, and then check for a number _higher_ than that.  I would assume that you're reying to include all "4GB" systems, so you want to thus check `>= 3725` ?

Reply:
The [correct computation](https://github.com/zulip/zulip/pull/28755#discussion_r1475086024) is 4 GB ⋅ (1000³ GB/B) / (1024² MiB/B) ≈ 3815 MiB.

Reply:
Yeah this is the result of my waffling a bit on what to propose. Adjusted the comments.